aikau/src/main/resources/alfresco/navigation/TreeStore.js

```diff
@@ -57,21 +100,66 @@ define(["dojo/_base/declare",
          var deferred = new Deferred();
-         var config = {
-            url: this.target + object.path,
-            query: this.targetQueryObject,
-            method: "GET",
-            deferred: deferred,
-            parent: object,
-            successCallback: this.onChildRequestSuccess,
-            failureCallback: this.onChildRequestFailure,
-            callbackScope: this
-         };
-         this.serviceXhr(config);
+         if (this.publishTopic)
+         {
+            var responseTopic = this.generateUuid();
+            var subscriptionHandles = [];
+
+            if (!this.publishPayload)
+            {
+               this.publishPayload = {};
+            }
+            this.publishPayload.alfResponseTopic = responseTopic;
+            this.publishPayload.path = object.path;
+
+            subscriptionHandles.push(this.alfSubscribe(responseTopic + "_SUCCESS", lang.hitch(this, this.onChildren, deferred, parent, subscriptionHandles)));
+            subscriptionHandles.push(this.alfSubscribe(responseTopic, lang.hitch(this, this.onChildren, deferred, object, subscriptionHandles)));
+            this.alfPublish(this.publishTopic, this.publishPayload, this.publishGlobal);
+         }
+         else
+         {
+            var config = {
+               url: this.target + object.path,
+               query: this.targetQueryObject,
+               method: "GET",
+               deferred: deferred,
+               parent: object,
+               successCallback: this.onChildRequestSuccess,
+               failureCallback: this.onChildRequestFailure,
+               callbackScope: this
+            };
+            this.serviceXhr(config);
+         }
          return deferred;
       },
       
       /**
+       * Handles child data provided when a [publishTopic]{@link module:alfresco/navigation/TreeStore#publishTopic} 
+       * has been configured to request child data.
+       * 
+       * @instance
+       * @param {object} deferred The deferred object to resolve with the child data
+       * @param {object} parent The parent node in the tree
+       * @param {object[]} subscriptionHandles Subscription handles created for the request that need to be removed
+       * @param {object} payload The payload containing the child data.
+       * @since 1.0.39
+       */
+      onChildren: function alfresco_navigation_TreeStore__onChildren(deferred, parent, subscriptionHandles, payload) {
+         this.alfUnsubscribeSaveHandles(subscriptionHandles);
+         array.forEach(payload.response.items, lang.hitch(this, this.updateChild, parent));
+         
+         // If required, filter the items based on their paths...
+         var filteredResponse = payload.response.items;
+         if (this.filterPaths !== null)
+         {
+            filteredResponse = array.filter(payload.response.items, lang.hitch(this, this.filterChildren));
```

Comment:
No point changing this, but could have been written `array.filter(...items, this.filterChildren, this)` without needing to invoke the extra `lang.hitch` function.